### PR TITLE
feat: account for rule creation time in performance reports

### PR DESCRIPTION
**Pull request by snitin315**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[x] Add something to the core
[ ] Other, please explain:

### What changes did you make? (Give an overview)

Fix https://github.com/eslint/eslint/issues/15952

### Before
Screenshot 2022-06-10 at 7 19 59 PM

### After
Screenshot 2022-06-10 at 7 20 44 PM

#### Is there anything you'd like reviewers to focus on?

Not sure if it should be `feat` or `fix` 😕 .

### CODE REVIEW

**Review by nzakas — CHANGES_REQUESTED** [review]
Code LGTM. I think we should also document this change to make sure it’s clear that the numbers represent creation+execution time.

File: docs/src/developer-guide/working-with-rules.md [hunk 1]
```diff
@@ -732,7 +732,7 @@ Performance budget ok:  1443.736547ms (limit: 3409.090909090909ms)
 
 ### Per-rule Performance
 
-ESLint has a built-in method to track performance of individual rules. Setting the `TIMING` environment variable will trigger the display, upon linting completion, of the ten longest-running rules, along with their individual running time and relative performance impact as a percentage of total rule processing time.
+ESLint has a built-in method to track performance of individual rules. Setting the `TIMING` environment variable will trigger the display, upon linting completion, of the ten longest-running rules, along with their individual running time (rule creation + rule execution) and relative performance impact as a percentage of total rule processing time (rule creation + rule execution).
```
Comment (snitin315) [inline docs/src/developer-guide/working-with-rules.md:736]: This is the main change, the rest of the file got formatted with prettier 😅

**Comment by mpsijm** [comment]
Thanks for implementing this! :smile: For reference, I will share the difference in timing below, for the rules of `eslint-plugin-tailwindcss` as I mentioned in #15952. Both measurements use `eslint-plugin-tailwindcss@3.5.0` (which is before my performance optimizations for that plugin got merged).

Before (with `eslint` set to `v8.17.0`):
```
$ TIMING=1 time npx eslint .
Rule                                           | Time (ms) | Relative
:----------------------------------------------|----------:|--------:
tailwindcss/enforces-shorthand                 |  2474.367 |    70.6%
tailwindcss/no-custom-classname                |   487.288 |    13.9%
tailwindcss/no-contradicting-classname         |   297.888 |     8.5%
tailwindcss/classnames-order                   |   211.153 |     6.0%
tailwindcss/migration-from-tailwind-2          |    19.401 |     0.6%
tailwindcss/enforces-negative-arbitrary-values |    16.575 |     0.5%

27.23user 0.84system 0:18.80elapsed 149%CPU (0avgtext+0avgdata 648716maxresident)k
0inputs+16outputs (0major+235883minor)pagefaults 0swaps

```

After (with `eslint` set to branch `fix/include-time-to-create-rule`):
```
$ TIMING=1 time npx eslint .
Rule                                           | Time (ms) | Relative
:----------------------------------------------|----------:|--------:
tailwindcss/classnames-order                   |  7172.373 |    50.6%
tailwindcss/no-custom-classname                |  2690.234 |    19.0%
tailwindcss/enforces-shorthand                 |  2555.669 |    18.0%
tailwindcss/no-contradicting-classname         |  1698.543 |    12.0%
tailwindcss/enforces-negative-arbitrary-values |    32.085 |     0.2%
tailwindcss/migration-from-tailwind-2          |    23.784 |     0.2%

26.98user 0.87system 0:18.52elapsed 150%CPU (0avgtext+0avgdata 598276maxresident)k
0inputs+16outputs (0major+225659minor)pagefaults 0swaps
```

Note how the CPU-time did not change (~27 seconds), but the total execution times for some rules change massively. I'd say, for this MR: mission accomplished :smile:

**Review by mpsijm — APPROVED** [review]

File: docs/src/developer-guide/working-with-rules.md [hunk 2]
```diff
@@ -346,18 +346,19 @@ Best practices for fixes:
...
+            { foo: 1 }
+        )(
+            // or
 
-        // or
-
-        ({ "foo": 1 })
+            { foo: 1 }
+        ));
         ```
```
Comment (mdjermanovic) [inline docs/src/developer-guide/working-with-rules.md:362]: This example looks broken now. What do you think about reverting the formatting changes, as they created a huge diff that is unrelated to this fix? I'm also not sure if this is formatting we would like to enforce, like 3 spaces after `*`, trailing commas in js code blocks, etc.

Comment (snitin315) [reply docs/src/developer-guide/working-with-rules.md:362]: Yeah, I will revert this formatting change. This got automatically formatted maybe I need to update my vscode extension settings.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Review by nzakas — APPROVED** [review]
LGTM.
